addons/sale/models/payment_transaction.py

```diff
@@ -36,17 +36,24 @@ def _compute_sale_order_ids_nbr(self):
 
     def _set_pending(self, state_message=None):
         """ Override of payment to send the quotations automatically. """
+        previous_states = {tx.id: tx.state for tx in self}
         super(PaymentTransaction, self)._set_pending(state_message=state_message)
 
-        for record in self:
-            sales_orders = record.sale_order_ids.filtered(lambda so: so.state in ['draft', 'sent'])
-            sales_orders.filtered(lambda so: so.state == 'draft').with_context(tracking_disable=True).write({'state': 'sent'})
-
-            if record.acquirer_id.provider == 'transfer':
-                for so in record.sale_order_ids:
-                    so.reference = record._compute_sale_order_reference(so)
-            # send order confirmation mail
-            sales_orders._send_order_confirmation_mail()
+        # When using webhooks, this function can be called two times (when the user comes back and
+        # when the webhooks notify the db). To avoid duplicates, check if the transaction is not
+        # already pending
+        for tx in self:
+            if previous_states[tx.id] != 'pending' and self.state == 'pending':
```

Comment:
for tx in self:
    self.state == 'pending'

:eyes:
(Not for this task, but this override of `_set_pending` seems quite crappy, probably a good place for further cleanups :D)